asv/benchmark.py

```diff
@@ -704,16 +706,35 @@ def __init__(self, name, func, attr_sources):
         self.unit = "bytes"
 
     def run(self, *param):
-        # We can't import asizeof directly, because we haven't loaded
-        # the asv package in the benchmarking process.
-        path = os.path.join(
-            os.path.dirname(__file__), 'extern', 'asizeof.py')
-        asizeof = importlib.machinery.SourceFileLoader('asizeof', path).load_module()
+        if ON_PYPY:
+            raise NotImplementedError("asizeof doesn't work on pypy")
+            return
+
+        def import_asizeof():
+            """Import asizeof, searching system Pythons in PATH."""
+            for path in os.environ.get("PATH", "").split(os.pathsep):
+                python_path = os.path.join(path, "python")
+                if os.path.isfile(python_path) and os.access(python_path, os.X_OK):
+                    cand_path = Path(python_path).parent.parent / "lib"
+                    asizeof_paths = [x for x in cand_path.rglob("asizeof.py")]
+                    if len(asizeof_paths) > 0:
+                        # FIXME: This will return the first path found
+                        asizeof = importlib.machinery.SourceFileLoader(
+                            "asizeof", str(asizeof_paths[0])
+                        ).load_module()
+                        return asizeof
+            return NotImplementedError("asizeof not found anywhere")
```

Comment:
This is a horrible hack, mostly because of persistent CI failures. A much nicer approach is to ensure either that the `virtualenv`, `mamba`, and `conda` environments always adds `pympler`, or even cleaner and less obtrusively, the approach in https://github.com/airspeed-velocity/asv/pull/1234/commits/c1245ccc34e3bec18c2cbb041402e6de414ce96e

Basically:
```python
        try:
            from pympler.asizeof import asizeof
        except ImportError:
            try:
                subprocess.run([sys.executable, "-m", "pip", "install", "pympler"], check=True)
                from pympler.asizeof import asizeof
            except (subprocess.CalledProcessError, ImportError) as e:
                raise NotImplementedError(f"Failed to install pympler: {e}")
```

Which works locally but fails on the CI persistently with `versions none` from `pip install`.